### ISSUE TITLE
[HOLD for payment 2024-11-13] [$250] Web - Messages - Unread marker is not displayed after the messages are received

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#44007]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.85-0
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:** https://expensify.testrail.io/index.php?/tests/view/4632530
**Email or phone of affected tester (no customers):**  test.iris.santos+17062024@gmail.com
**Issue reported by:** Applause - Internal Team

## Action Performed:
Preconditions:
- Navigate to https://staging.new.expensify.com/
in any browser
- Log in with an account
- Log in with a secondary account in another device or incognito tab.
1. Open a 1:1 conversation with a user you have access with.
2. While viewing the 2 conversations at the same time
3. Turn off the internet connection in the device
4. As the secondary user - send a message to the account you're logged in the testing device
5. Turn on the internet connection
6. See the message sent from the other account

## Expected Result:
The unread marker (green line and 'New' label) is displayed in the conversation after the messages are received

## Actual Result:
The unread marker (green line and 'New' label) is NOT displayed in the conversation after the messages are received

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/Expensify/App/assets/78819774/5034a743-743b-4579-be4e-d53be27a2a04

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01fbab0fd3569bf04f</li>
        <li>Upwork Job ID: 1823480826689774019</li>
        <li>Last Price Increase: 2024-09-03</li>
<li>Automatic offers: </li>
<ul>
<li>brunovjk | Reviewer | 103816429</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @alexpensify</details>

**Comment by alexpensify** [comment]
I'll review this one soon.

**Comment by alexpensify** [comment]
Still on my testing radar

**Comment by alexpensify** [comment]
Other GHs and customer tasks have been a higher priority, I'll get to this one soon.

**Comment by alexpensify** [comment]
After reviewing these steps, I don't think it's a dupe but there is a known issue with the `new message` banner when online. It's being addressed here: https://github.com/Expensify/App/issues/43486. I'm going to put this GH on hold and then test it again after #43486 in Production. Setting a reminder to check next week. 

--- 

## Heads up, I will be offline until Monday, July 8, 2024, and will not actively watch over this GitHub during that period.

If this GitHub needs an urgent update, please ask for help in the #expensify-open-source Slack Room. If it can wait, I'll continue the review process when I return online. Thanks!

**Comment by alexpensify** [comment]
Still on hold for https://github.com/Expensify/App/issues/43486

**Comment by alexpensify** [comment]
No update, on hold for https://github.com/Expensify/App/issues/43486

**Comment by alexpensify** [comment]
@bernhardoj and @shubham1206agra - I believe this one is resolved by the other GH, but let me know if you disagree. Thanks!

**Comment by bernhardoj** [comment]
The PR for https://github.com/Expensify/App/issues/43486 is actually reverted and the new PR hasn't been merged yet.

**Comment by alexpensify** [comment]
Thanks, I missed that there was a revert. Ok, keeping this one on hold.

**Comment by alexpensify** [comment]
Weekly Update: On hold for https://github.com/Expensify/App/issues/43486

**Comment by alexpensify** [comment]
@bernhardoj and @shubham1206agra can I get your feedback if we should close this one or move forward with the next steps? Thanks!

**Comment by trjExpensify** [comment]
Chat bug, removing from wave-collect.

**Comment by alexpensify** [comment]
SO related to this Process: https://stackoverflowteams.com/c/expensify/questions/14418/14456#14456

@lanitochka17 - can you please retest this one? I added the label too. Thanks!

**Comment by MelvinBot** [comment]
This has been labelled "Needs Reproduction". Follow the steps here: https://stackoverflowteams.com/c/expensify/questions/16989

**Comment by kavimuru** [comment]
Bug is still reproducible. 

https://github.com/user-attachments/assets/e9ef1309-c100-4f71-a45f-626b42158d66

**Comment by alexpensify** [comment]
Thanks, I'm marking this one as external, and we are open for proposals here.

**Comment by alexpensify** [comment]
Waiting for proposals

**Comment by brunovjk** [comment]
Bumped on [Slack](https://expensify.slack.com/archives/C01GTK53T8Q/p1724066974065989) for proposals

**Comment by chrispader** [comment]
I just wanted to confirm, that the "unread" marker and "New" label are actually supposed to be shown when the user receives messages while still in the chat.

The marker should only be visible when the receiving device was offline, correct? Not when the sending device was offline, but the receiving device was online...

cc @alexpensify

**Comment by chrispader** [comment]
Also i'm curious about the provided repro videos:

In the [video in the issue description](https://private-user-images.githubusercontent.com/78819774/341129872-5034a743-743b-4579-be4e-d53be27a2a04.mp4?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQzMjczMDksIm5iZiI6MTcyNDMyNzAwOSwicGF0aCI6Ii83ODgxOTc3NC8zNDExMjk4NzItNTAzNGE3NDMtNzQzYi00NTc5LWJlNGUtZDUzYmUyN2EyYTA0Lm1wND9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODIyVDExNDMyOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTE3OTE1MjcwMDI0NTE0MGE2MjM2OWQzNjk1YTQ0Y2Y4NWNlM2VlODZiMDg0ZjNhODA4MDZiNjQ1ZDUxYzg5OGQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.CGq_KPeSecwYMdvlegqUVfNbG9ENBC5aWjgbt0w5KBU), the WiFi is turned off, but only the second (right) tab goes offline. When WiFi is turned on again, the **"New messages" popup** at the top of the chat appears, but no "unread" marker.

In [this repro video](https://github.com/Expensify/App/issues/44007#issuecomment-2286035303), only the internet connection of the receiving device is turned off. When the tab goes back online again, there is **NO "New messages" popup**.

Therefore, **what is the expected behavior**? Whenever the (receiving) user goes offline and comes back online, there should be both...

- a "New messages" popup at the top and
- a "unread" marker

... or nothing at all? When the user is online, there should be no indication of unread messages, right?

cc @alexpensify

**Comment by chrispader** [comment]
I cannot consistently reproduce [CASE 1], i only see the "New messages" banner sometimes.

Here are some of my screen recordings with different scenarios on both DEV and STG.

DEV: [VIDEO](https://drive.google.com/file/d/1xCiK2jkLDOTUz8Etlgb3AcbSwxTFkwIm/view?usp=share_link)
STG: [VIDEO](https://drive.google.com/file/d/133vBwhjFJphFzZdXSQgf5TMsXcasb-2a/view?usp=sharing)

**Comment by alexpensify** [comment]
@brunovjk - when you get a chance, can you please review the questions? Thanks!

**Comment by brunovjk** [comment]
Sure, I'll get back here ASAP.

**Comment by brunovjk** [comment]
Apologies for the delay. I'll reach out to an internal to help confirm the expected behavior.

🎀👀🎀

**Comment by brunovjk** [comment]
@tylerkaraszewski, could you help us confirm the expected behavior in this scenario? There's some confusion regarding the "New messages" popup and "unread" markers. Please take a look at @chrispader's [comments](https://github.com/Expensify/App/issues/44007#issuecomment-2304512506) for more context. Thank you :D

**Comment by alexpensify** [comment]
## Next Steps

We need an internal review of this proposal 

## Heads up, I will be offline until Tuesday, September 3, 2024, and will not actively watch over this GitHub during that period. 

If anything urgent is needed here, please ask for help in the #expensify-open-source Slack Room-- thanks!

**Comment by brunovjk** [comment]
Excuse me @tylerkaraszewski, did you have a chance to look at **chrispader's** [comments](https://github.com/Expensify/App/issues/44007#issuecomment-2304512506)? Thanks.

**Comment by tylerkaraszewski** [comment]
@roryabraham - I picked you here because I feel like you'd know the answer.

I don't actually know what we expect the behavior here to be when you've been offline but come back online while the new messages are in the focused foreground window. I feel like we should not see a notification badge on the conversation, because the conversation is already loaded, but we *should* see the unread marker line, as this can mean many messages are delivered at once, and that line is used to indicate where you'd been in the conversation previously.

This is fairly nuanced though so I'd love a second opinion.

**Comment by roryabraham** [comment]
I'd agree with that @tylerkaraszewski, but TBC I'm not sure that this is a "bug" per-se, but a new feature.

Currently, the Unread Indicator only appears when you open a chat after having a different one open. But I agree that it makes sense to also show it if you've been offline and come back online, even if you've had the chat open the whole time 👍🏼

**Comment by chrispader** [comment]
Alright, i'm going to set up a PR for this later today, if that's alright?

Going for this approach:

- Only show a notification badge when the user is not currently in the report
- Show both "New messages" popup at the top and the unread marker line when (multiple) new messages are received

**Comment by brunovjk** [comment]
> Alright, i'm going to set up a PR for this later today, if that's alright?
> 
> Going for this approach:
> 
> * Only show a notification badge when the user is not currently in the report
> * Show both "New messages" popup at the top and the unread marker line when (multiple) new messages are received

Sounds good to me, what do you think @roryabraham?

**Comment by roryabraham** [comment]
@chrispader to clarify, are you proposing to show the new message indicator when _any_ new messages appear in a chat you're currently viewing?

I'd think that we'd only want to show the new message indicator if you were offline and come back online to new messages you haven't received. Perhaps we should discuss in slack though

**Comment by chrispader** [comment]
> Show both "New messages" popup at the top and the unread marker line when (multiple) new messages are received

@roryabraham Sorry for the confusion. Should have added "...while offline" at the end. I think we have the same approach in mind!

I wanted the "New messages" popup and unread marker to only be displayed when the user was offline or in another report

**Comment by brunovjk** [comment]
Hey @chrispader, just checking in—do you have everything needed to keep things moving forward, or is there anything I can assist with? Let me know if you need any help! Thank you :D

**Comment by roryabraham** [comment]
sounds good @chrispader 👍🏼

**Comment by chrispader** [comment]
@brunovjk yes sorry for the delay. i have everything i need, was busy with other tasks until now, but going to fix this soon!

**Comment by brunovjk** [comment]
Not overdue, **chrispader** [working on it](https://github.com/Expensify/App/issues/44007#issuecomment-2330773855).

**Comment by alexpensify** [comment]
@chrispader - any update here? Thanks!

**Comment by chrispader** [comment]
Sorry for the delay! I'm going to give a detailed update on this tmrw!

**Comment by alexpensify** [comment]
No overdue, we are waiting for an update here

**Comment by chrispader** [comment]
Created a PR for this issue in https://github.com/Expensify/App/pull/49480, but this is still WIP. Will finish by tmrw!

**Comment by alexpensify** [comment]
Awesome, thanks @chrispader for the update!

**Comment by chrispader** [comment]
PR is ready for review! cc @brunovjk @alexpensify

**Comment by alexpensify** [comment]
Awesome, the PR is moving along

**Comment by alexpensify** [comment]
Update: PR is still under review

**Comment by alexpensify** [comment]
@chrispader - it looks like the PR is stalling based on the tests. Is that a good summary?

**Comment by chrispader** [comment]
@alexpensify yes correct, so in the PR i assumed, the 20% decrease in render time is caused by the extra check for "offline messages"

**Comment by alexpensify** [comment]
Thanks for the update! Is there anything we can do to move forward, like starting a discussion with this team in the Slack Open Source room?

**Comment by alexpensify** [comment]
## Heads up, I will be offline until Tuesday, October 22, 2024, and will not actively watch over this GitHub during that period.

If this GitHub requires an urgent update, please ask for help in the #expensify-open-source Slack Room. If it can wait, I'll review it when I return online.

**Comment by brunovjk** [comment]
Update: PR is still under review

**Comment by alexpensify** [comment]
Thank you for the update!

**Comment by brunovjk** [comment]
Update: PR on staging.

**Comment by brunovjk** [comment]
The PR is failing due to its own issue [#49480 - Failing with #44007](https://github.com/Expensify/App/issues/51953#issuecomment-2456775928). This is strange because I tested it just now, and everything seems normal to me:

https://github.com/user-attachments/assets/2a3bc082-7585-4ea6-8e7a-acac6dd2a5f3

@chrispader Do you think we missed something?

**Comment by chrispader** [comment]
> The PR is failing due to its own issue [#49480 - Failing with #44007](https://github.com/Expensify/App/issues/51953#issuecomment-2456775928). This is strange because I tested it just now, and everything seems normal to me:
> 
>  Screen.Recording.2024-11-05.at.09.37.17.mov 
> @chrispader Do you think we missed something?

DId you test this change on `staging`? I don't think there's anything wrong with this PR

**Comment by brunovjk** [comment]
Exactly, I had only tested PR. But in staging it also looks good to me:

https://github.com/user-attachments/assets/ed00f46f-5051-4ddc-9d5c-65454e2f3710

**Comment by brunovjk** [comment]
I forgot to point out that our [PR](https://github.com/Expensify/App/pull/49480) was missing **QA Steps** 😬 I don't know if that has anything to do with it.

**Comment by chrispader** [comment]
Since this has been merged to production, i think the deploy blocker is obsolete? @brunovjk

**Comment by brunovjk** [comment]
I think so, but I'm not sure what happened here :/ I'll ask on Slack later.

**Comment by brunovjk** [comment]
# BugZero Checklist:

- [x] **[Contributor]** Classify the bug:

<details>
<summary>Bug classification</summary>


Source of bug:
  - [x] 1a. Result of the original design (eg. a case wasn't considered)
  - [ ] 1b. Mistake during implementation
  - [ ] 1c. Backend bug
  - [ ] 1z. Other:

Where bug was reported:
  - [x] 2a. Reported on production
  - [ ] 2b. Reported on staging (deploy blocker)
  - [ ] 2c. Reported on a PR
  - [ ] 2z. Other:

Who reported the bug:
  - [ ] 3a. Expensify user
  - [ ] 3b. Expensify employee
  - [ ] 3c. Contributor
  - [ ] 3d. QA
  - [x] 3z. Other: Applause - Internal Team

</details>

- [ ] **[Contributor]** The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake.

    Link to comment:

- [ ] **[Contributor]** If the regression was CRITICAL (e.g. interrupts a core flow) A discussion in [#expensify-open-source](https://app.slack.com/client/E047TPA624F/C01GTK53T8Q) has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner.

    Link to discussion:

- [x] **[Contributor]** If it was decided to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps using the template below to ensure the same bug will not reach production again.

## Regression Test Proposal
### Precondition:
- User logged into the platform with two separate accounts on different devices or browser windows.

### Test:
1. On device A, turn off the internet connection.
2. On device B, send a message to the account on device A.
3. On device A, turn on the internet connection.
4. Verify if the unread marker (green line and 'New' label) is displayed after receiving the message.

Do we agree 👍 or 👎

**Comment by alexpensify** [comment]
Payouts due: 2024-11-13

- [ ] Contributor: @chrispader payment handled via vendor 
- [ ] Reviewer: $250 @brunovjk 

Upwork job is [here](https://www.upwork.com/jobs/~021823480826689774019).

**Comment by alexpensify** [comment]
All set, I've completed the payment to @brunovjk via Upwork. I'll work on the QA test prep later this week.

**Comment by alexpensify** [comment]
Not overdue

**Comment by brunovjk** [comment]
**Update**: Not overdue, working on the QA test. @alexpensify, do you think I can help in any way?

**Comment by alexpensify** [comment]
No, sorry, other GHs have been a priority. I'll get to the regression test soon.

**Comment by alexpensify** [comment]
With the slower week, I'll be working on the last step soon.

**Comment by alexpensify** [comment]
Closing - all set here
